File: mycollab-scheduler-ondemand/src/test/java/com/mycollab/schedule/CronTriggerTest.java
package com.mycollab.schedule;

import org.junit.Assert;
import org.junit.Test;
import org.quartz.CronExpression;

import java.text.ParseException;
import java.util.Date;
import java.util.GregorianCalendar;

public class CronTriggerTest {

    @Test
    public void testTriggerEveryBeginningDay() throws ParseException {
        CronExpression expr = new CronExpression("0 0 0 * * ?");
        GregorianCalendar calendar = new GregorianCalendar(2014, 7, 3, 9, 30);
        Date validDate = expr.getNextValidTimeAfter(calendar.getTime());
        Assert.assertEquals(new GregorianCalendar(2014, 7, 4, 0, 0, 0).getTime(), validDate);
    }

    @Test
    public void testTriggerEveryMinutes() throws ParseException {
        CronExpression expr = new CronExpression("0 * * * * ?");
        GregorianCalendar calendar = new GregorianCalendar(2014, 7, 3, 9, 30);
        Date validDate = expr.getNextValidTimeAfter(calendar.getTime());
        Assert.assertEquals(new GregorianCalendar(2014, 7, 3, 9, 31, 0).getTime(), validDate);
    }
}

File: mycollab-scheduler-ondemand/src/test/java/com/mycollab/schedule/CronTriggerTest.kt
package com.mycollab.schedule

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.quartz.CronExpression
import java.text.ParseException
import java.util.*

class CronTriggerTest {

    @Test
    @Throws(ParseException::class)
    fun testTriggerEveryBeginningDay() {
        val expr = CronExpression("0 0 0 * * ?")
        val calendar = GregorianCalendar(2014, 7, 3, 9, 30)
        val validDate = expr.getNextValidTimeAfter(calendar.time)
        Assertions.assertEquals(GregorianCalendar(2014, 7, 4, 0, 0, 0).time, validDate)
    }

    @Test
    @Throws(ParseException::class)
    fun testTriggerEveryMinutes() {
        val expr = CronExpression("0 * * * * ?")
        val calendar = GregorianCalendar(2014, 7, 3, 9, 30)
        val validDate = expr.getNextValidTimeAfter(calendar.time)
        Assertions.assertEquals(GregorianCalendar(2014, 7, 3, 9, 31, 0).time, validDate)
    }
}
